File: Link List/medium/Reverse LL.cpp
    ListNode* addBack(ListNode* head, int val){
        ListNode* newHead = new ListNode(val);
        newHead->next = head;
        return newHead;
    }

    ListNode* reverseList(ListNode* head) {

        if(head == NULL || head->next == nullptr){
            return head;
        }
        

        ListNode* temp = head->next;
        head->next = nullptr;

        while(temp != nullptr){
            int val = temp->val;
            head = addBack(head, val);
            ListNode * prev = temp;
            temp = temp->next;
            prev->next = nullptr;
            delete prev;
        }

        return head;
    }

File: Link List/medium/ReverseLL.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        it = head
        back = None
        ft = None

        if head == None or head.next == None:
            return head

        while it != None:
            ft = it.next
            it.next = back
            back = it
            it = ft

        return back
